isis/tests/FunctionalTestsChan1m32isis.cpp

```diff
@@ -0,0 +1,558 @@
+  ASSERT_DOUBLE_EQ(bandbin["FilterNumber"][255].toDouble(), 260);
+
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][0].toDouble(), 1);
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][64].toDouble(), 65);
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][128].toDouble(), 129);
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][255].toDouble(), 256);
+
+  // Kernels Group
+  PvlGroup &kernel = isisLabel->findGroup("Kernels", Pvl::Traverse);
+  ASSERT_EQ(int(kernel["NaifFrameCode"]), -86520);
+
+  Histogram *hist = cube.histogram();
+
+  ASSERT_DOUBLE_EQ(hist->Average(), 15.58169779027077);
+  ASSERT_DOUBLE_EQ(hist->Sum(), 47368.361282423139);
+  ASSERT_EQ(hist->ValidPixels(), 3040);
+  ASSERT_DOUBLE_EQ(hist->StandardDeviation(), 2.2696592481066249);
+
+  delete hist;
+}
+
+
+TEST(Chan1m32Isis, Chan1m32IsisTestFowardDescending) {
+  QTemporaryDir prefix;
+  QString cubeFileName = prefix.path() + "/chan1m32isisTEMP.cub";
+  QVector<QString> args = {"from=data/chan1m32isis/forwardDescending/M3G20081129T171431_V03_L1B_cropped.LBL",
+                           "loc=data/chan1m32isis/forwardDescending/M3G20081129T171431_V03_LOC_cropped.IMG",
+                           "obs=data/chan1m32isis/forwardDescending/M3G20081129T171431_V03_OBS_cropped.IMG",
+                           "to=" + cubeFileName };
+
+  UserInterface options(APP_XML, args);
+  try {
+    chan1m32isis(options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to ingest Chandrayaan image: " << e.toString().toStdString().c_str() << std::endl;
+  }
+  Cube cube(cubeFileName);
+  Pvl *isisLabel = cube.label();
+
+  // Dimensions Group
+  PvlGroup &dimensions = isisLabel->findGroup("Dimensions", Pvl::Traverse);
+  ASSERT_EQ(int(dimensions["Samples"]), 304);
+  ASSERT_EQ(int(dimensions["Lines"]), 5);
+  ASSERT_EQ(int(dimensions["Bands"]), 3);
+
+  // Pixels Group
+  PvlGroup &pixels = isisLabel->findGroup("Pixels", Pvl::Traverse);
+  ASSERT_EQ(pixels["Type"][0].toStdString(), "Real");
+  ASSERT_EQ(pixels["ByteOrder"][0].toStdString(), "Lsb");
+  ASSERT_EQ(double(pixels["Base"]), 0.0);
+  ASSERT_EQ(double(pixels["Multiplier"]), 1.0);
+
+  // Instrument Group
+  PvlGroup &inst = isisLabel->findGroup("Instrument", Pvl::Traverse);
+  ASSERT_EQ(inst["SpacecraftName"][0].toStdString(), "CHANDRAYAAN-1");
+  ASSERT_EQ(inst["InstrumentId"][0].toStdString(), "M3" );
+  ASSERT_EQ(inst["TargetName"][0].toStdString(), "MOON" );
+  ASSERT_EQ(inst["SpacecraftClockStartCount"][0].toStdString(), "2/1531046.542" );
+  ASSERT_EQ(inst["SpacecraftClockStopCount"][0].toStdString(), "2/1531047.050" );
+  ASSERT_DOUBLE_EQ(double(inst["LineExposureDuration"]), 101.76);
+  ASSERT_EQ(inst["StartTime"][0].toStdString(), "2008-11-29T17:14:29.729807" );
+  ASSERT_EQ(inst["StopTime"][0].toStdString(), "2008-11-29T17:14:30.238607" );
+  ASSERT_EQ(inst["SpacecraftYawDirection"][0].toStdString(), "FORWARD" );
+  ASSERT_EQ(inst["OrbitLimbDirection"][0].toStdString(), "DESCENDING" );
```

Comment:
Similarly outside of the default I don't know if it's worth checking every keyword here. This test really only seems to care that `SpacecraftYawDirection` is FORWARD and `OrbitLimbDirection` is DESCENDING. For tests that are explicitly looking to test those features I don't think it's super important to test all keywords in the instrument group.